### PR TITLE
Stax

### PR DESCRIPTION
**Pull request by PolyProgrammist**
# Checklist

- [ ] App update process has been followed 
- [x] Target branch is `develop` 
- [x] Application version has been bumped

### CODE REVIEW

**Review by frol — APPROVED** [review]
I reviewed this massive PR and verified that everything works as expected. @dj8yfo has also checked that it works on Nano devices.

@agrojean-ledger Please, review and merge it

File: Cargo.toml
Comment (agrojean-ledger) [inline Cargo.toml]: Version should be bumped here.

**Comment by agrojean-ledger** [comment]
@PolyProgrammist @frol are you sure you need `NbglGenericReview` where you are using it ?

The generic review should always be used as a last resort.

Maybe the  [choice use case](https://github.com/LedgerHQ/ledger-device-rust-sdk/pull/173) would allow you to generate the screens you need ? 

It is now available in the latest version of the Rust SDK : [1.13.0](https://crates.io/crates/ledger_device_sdk)

You would be able to add screens like that for instance : 

TestUseCaseChoice

The API allows to add (or not) status text when the user clicks on one of the buttons.

File: src/app_ui/sign/widgets.rs
Comment (agrojean-ledger) [inline src/app_ui/sign/widgets.rs:48]: What kind of API do you need here ? Something that just displays a spinner with some text ? But when should it return ?

File: Cargo.toml
Comment (frol) [reply Cargo.toml]: 2.0.0 has not been released/published yet, I don't see the reason to bump it.

File: src/app_ui/sign/widgets.rs
Comment (frol) [reply src/app_ui/sign/widgets.rs:48]: Yes, we would need some loading indicator. After showing it, we dive into receiving a stream of data and deserializing it on the go.

https://github.com/LedgerHQ/app-near-new/blob/dc6e6a2a9d68186a82cdcb964679528ff792f9a9/src/handlers/sign_tx.rs#L56

**Comment by frol** [comment]
> are you sure you need NbglGenericReview where you are using it ?

@agrojean-ledger Yes, I am absolutely sure. We use these review screens independently and on the fly while the data is getting streamed on the fly and we don't have the memory to load all the choices upfront.

Thank you for the review, but for god's sake, please, stop doing this "and one more thing". Please, publish this app finally.

File: Cargo.toml
Comment (agrojean-ledger) [reply Cargo.toml]: We have already deployed it on our test provider, so you need to bump it, sorry.

**Comment by agrojean-ledger** [comment]
> > are you sure you need NbglGenericReview where you are using it ?
> 
> @agrojean-ledger Yes, I am absolutely sure. We use these review screens independently and on the fly while the data is getting streamed on the fly and we don't have the memory to load all the choices upfront.
> 
> Thank you for the review, but for god's sake, please, stop doing this "and one more thing". Please, publish this app finally.

@frol @PolyProgrammist I suggest you look at the `NbglChoice` that was just added to the SDK, I think it would be more fitting where you try to use it.

File: src/app_ui/sign/widgets.rs
Comment (agrojean-ledger) [reply src/app_ui/sign/widgets.rs:48]: So it needs to be a non-blocking call, right ?

I'm asking so we can improve the SDK, not to annoy you.

**Comment by frol** [comment]
> I suggest you look at the NbglChoice that was just added to the SDK, I think it would be more fitting where you try to use it.

What is exactly wrong with our implementation from the user perspective that you suggest us to go into refactoring again?

**Comment by agrojean-ledger** [comment]
> > I suggest you look at the NbglChoice that was just added to the SDK, I think it would be more fitting where you try to use it.
> 
> What is exactly wrong with our implementation from the user perspective that you suggest us to go into refactoring again?

It was just a suggestion. Anyway, I don't understand your code enough to know if you could replace the generic reviews by choice screens.

Bump the version of the package and we'll be done with.

I will merge your PR but the app will only be deployed when it is requested by our product team.

File: src/app_ui/sign/widgets.rs
Comment (frol) [reply src/app_ui/sign/widgets.rs:48]: Yes, it needs to be non-blocking

Comment (agrojean-ledger) [reply src/app_ui/sign/widgets.rs:48]: Ok thx.

File: Cargo.toml
Comment (frol) [reply Cargo.toml]: @agrojean-ledger Do you suggest we bump it to 2.1.0?

Comment (agrojean-ledger) [reply Cargo.toml]: Yes, just bump the minor version :+1:

Comment (frol) [reply Cargo.toml]: Done

**Comment by agrojean-ledger** [comment]
@frol @PolyProgrammist The CI is KO, you need to fix it before we can merge.

**Comment by PolyProgrammist** [comment]
@agrojean-ledger thanks for the review. Could you please rerun CI?

**Comment by frol** [comment]
@agrojean-ledger Please, approve CI once again

**Comment by agrojean-ledger** [comment]
@frol @PolyProgrammist There's still one job that fails... you're almost there.

**Review by agrojean-ledger — APPROVED** [review]

**Pull request merged**
